### DISCUSSION TITLE
Question regarding RBAC

### DISCUSSION THREAD

**Discussion by khaidoan** [Budibase/budibase#14001] [Help] [answered]
I am building a flashcard application for my own needs, and perhaps open source it for other people to use for free.  I have watched a lot of your videos, but I did not find one regarding RBAC.  I understand that Budibase has Basic, Admin, and Power.  I have two questions.

1. I want people to be able to sign-up using Google.  Currently, I have to manually add the users to Budibase first before the user can sign-in.  Is there a way to avoid this?  I intend to have only 1 application (this flashcard application) deployed.  Is there a way to configure budibase so if the user is accessing my application using a specific domain, that the user is automatically granted access to this flashcard application with the Basic role.

2. In my case, the user should be able to create decks, add flashcards, invite other people to the platform by sharing decks.  The users should be able to delete their own cards and decks.  Users should not be able to search and see decks that were created by other users (and had not been marked as public).  Can you please make a video on how Budibase support RBAC?  Can I achieve my goal using Budibase?  Can you do a video building flashcard application using Budibase?  How can I go about accomplishing this?

Thank you!
Khai

**Comment by mikesealey** [comment, marked as answer]
Hi @khaidoan 

Budibase it set up to invite users to your app in the way you're currently doing (mentioned in point 1). There is a public API that could be used to develop the functionality for a a passerby to create an account without being invited. You can read more about it [here](https://docs.budibase.com/docs/public-api).

As for the design of the app, I've put something together as a rough-and-ready example for you, you can find that attached below.

In broad strokes: 
I've set up a table with the following columns:
- auto_id
- created_at
- created_by
- title
- description
- slide1
- slide2
- slide3
- public

You can then set up a form, which will be built for you if you select "form", and set to power
<img width="1083" alt="image" src="https://github.com/Budibase/budibase/assets/39269380/3c6bbfde-75d9-4d14-bcab-0baaba74c6a0">
<img width="415" alt="image" src="https://github.com/Budibase/budibase/assets/39269380/3bf0c7e2-d44e-49e0-9fb0-574c072a047d">


(you may need to drag and shuffle some column names around and hide some auto_columns in the form)
<img width="292" alt="image" src="https://github.com/Budibase/budibase/assets/39269380/1fb914e7-f25d-4e51-85d8-03e9c81de040">

You can then set up a Data Provider and a repeater to loop over the values of the table, and display data like the title, and description. You could also set filters on that data provider to only display rows that are public or created by the currently logged-in user.
![image](https://github.com/Budibase/budibase/assets/39269380/be5971bd-dd70-4246-ac8b-8cd9cea29cb9)

The container inside the repeater has an On Click action to navigate to a new screen, which takes the _id of the slides row in question...
![image](https://github.com/Budibase/budibase/assets/39269380/e2d98046-9774-49f1-aad2-6b418ed83e30)

... And that screen has a Data Provider fetching rows from the slides table that match the _id provided, and uses the values stored in them.
![image](https://github.com/Budibase/budibase/assets/39269380/0dc9121d-92c4-4ce9-895c-118cd4ff5e15)

You can read more about passing variables like row id in the URL in [this discussion here](https://github.com/Budibase/budibase/discussions/13796)

[14001-export-1719398254556.tar.gz](https://github.com/user-attachments/files/15986642/14001-export-1719398254556.tar.gz)
